pandas/core/indexes/base.py

```diff
@@ -5274,6 +5276,14 @@ def _add_logical_methods_disabled(cls):
 Index._add_comparison_methods()
 
 
+class IndexGroupbyGroups(dict):
```

Comment:
this should be defined in pandas/core/groupby/groupby.py and used on the dict returning functions, mainly ``groups`` and ``indices``

this is not user facing function and kind of obscures that this is from groupby

Reply:
actually, maybe best to actually put this in pandas.io.format.printing and name it PrettyDict(dict), see my other notes for where to actually use this.

Reply:
Sure, I'll move this. To your first point, are you saying a groups property should do something like this: `return PrettyDict(self.grouper.groups)`? @WillAyd mentioned had noted that he'd prefer to put this in a different place, as multiple calls to the method will create a new object every time.

Reply:
yes; thiat would be a centralized place

Reply:
The topic initially came up because instantiating in the `groups` method was failing this test:

```
def test_groups(self, df):
        grouped = df.groupby(['A'])
        groups = grouped.groups
        assert groups is grouped.groups  # caching works
```

@WillAyd Thoughts?

Reply:
What @jreback says here makes a lot of sense - go ahead with the move!

Reply:
Thanks, I'll remove this test as part of the next commit.